src/t_stream.c

```diff
@@ -1422,39 +1412,56 @@ int streamRangeHasTombstones(stream *s, streamID *start, streamID *end) {
     return 0;
 }
 
-/* Replies with a consumer group's current lag, that is the number of messages
- * in the stream that are yet to be delivered. In case that the lag isn't
- * available due to fragmentation, the reply to the client is a null. */
+/* Replies with a consumer group's current lag, which is the number of messages in the stream
+ * that are yet to be delivered. Additionally, it includes an entries-read field that indicates
+ * the number of messages currently read. In case that the lag or entries-read isn't available
+ * due to fragmentation, the reply to the client is null. */
 void streamReplyWithCGLag(client *c, stream *s, streamCG *cg) {
     int valid = 0;
     long long lag = 0;
 
-    if (!s->entries_added) {
-        /* The lag of a newly-initialized stream is 0. */
-        lag = 0;
-        valid = 1;
-    } else if (cg->entries_read != SCG_INVALID_ENTRIES_READ && !streamRangeHasTombstones(s, &cg->last_id, NULL)) {
-        /* No fragmentation ahead means that the group's logical reads counter
-         * is valid for performing the lag calculation. */
-        lag = (long long)s->entries_added - cg->entries_read;
+    /* Attempt to retrieve the group's last ID logical read counter. */
+    long long entries_read = streamEstimateDistance(s, cg, &cg->last_id);
+    if (entries_read != SCG_INVALID_ENTRIES_READ) {
+        /* A valid counter was obtained. */
+        lag = (long long)s->entries_added - entries_read;
```

Comment:
Estimate distance is just an estimate, not exact?

If it is exact, then we should rename the *estimate* functions to something more exact like *calculate*, *compute*, etc.